### PR TITLE
Fix std calculation using unbiased estimation in sharing stat mode.

### PR DESCRIPTION
**Pull request by cxxgtxy**

### CODE REVIEW

**Comment by cxxgtxy** [comment]
Anyone check it?

**Comment by itaicaspi-intel** [comment]
Hi @cxxgtxy,
Thanks for your PR! 
The bias correction looks correct but there is an issue with the types.
Seems like `self._count` should be first casted into float
After correcting this, the change looks ok.

**Comment by cxxgtxy** [comment]
HI @itaicaspi-intel.
In fact, I have a question about the cast, since self_count is float64                
self._count = tf.get_variable(
                    dtype=tf.float64,
                    shape=(),
                    initializer=tf.constant_initializer(epsilon),
                    name="count", trainable=False)

File: architectures/tensorflow_components/shared_variables.py [hunk 1]
```diff
@@ -41,7 +41,8 @@ def __init__(self, tuning_parameters, replicated_device, epsilon=1e-2, shape=(),
 
                 self._shape = shape
                 self._mean = tf.to_float(self._sum / self._count)
```
Comment (itaicaspi-intel) [inline architectures/tensorflow_components/shared_variables.py]: self._mean = self._sum / self._count

File: architectures/tensorflow_components/shared_variables.py [hunk 2]
```diff
@@ -41,7 +41,8 @@ def __init__(self, tuning_parameters, replicated_device, epsilon=1e-2, shape=(),
...
-                self._std = tf.sqrt(tf.maximum(tf.to_float(self._sum_squared / self._count) - tf.square(self._mean), 1e-2))
+                self._std = tf.sqrt(tf.maximum((tf.to_float(self._sum_squared) - self._count*tf.square(self._mean))
```
Comment (itaicaspi-intel) [inline architectures/tensorflow_components/shared_variables.py]: self._std = tf.sqrt(tf.maximum(self._sum_squared - self._count * tf.square(self._mean), epsilon)
                                    / tf.maximum(self._count - 1, 1))

**Comment by cxxgtxy** [comment]
Good job.

**Comment by cxxgtxy** [comment]
I have changed it as your suggestion.

**Comment by itaicaspi-intel** [comment]
Great! thanks for your help!

**Pull request merged**